cranelift/codegen/src/isa/x64/inst/mod.rs

```diff
@@ -342,6 +342,10 @@ pub enum Inst {
         is64: bool,
     },
 
+    /// Provides a way to tell the register allocator that the upcoming sequence of instructions
+    /// will overwrite `dst` so it should be considered as a `def`; use with care.
```

Comment:
I think we should have a bit more here to describe when this might be useful: basically, when we have a sequence of instructions whose register usages are nominally `mod`s, but such that the combination of operations creates a result that is independent of the initial register value. It's thus semantically a `def`, not a `mod`, when all the instructions are taken together, so we want to ensure the register is defined (its live-range starts) prior to the sequence to keep analyses happy.

It's worth briefly mentioning the alternative as well, namely a compound instruction that somehow encapsulates the others and reports its own defs/uses/mods; this adds complexity (the instruction list is no longer flat) and requires knowledge about semantics and initial-value independence anyway.

Reply:
Also, I think it might be good to choose a better name. In the spirit of Rust's `mem::uninitialized()`, can we have `XmmUninitializedValue` that defines a register to an uninitialized value?

Reply:
Not a huge fan of this; instead could we just xorps the destination register, in the place where it's used? (There's already special handling of xorps X X in regalloc.) The slow down caused by the additional instruction in the pipeline may be recouped by the speedup caused by the lane dependency clearing.

If we really really have to resort to a vcode instruction for this, please put it at the bottom of this list (there's a section for instructions that don't generate new code); Chris's name suggestion sounds good to me.